tests/testthat/_snaps/rank.md

```diff
@@ -1,3 +1,19 @@
+# `x` must not be `NULL` (#1823)
+
+    Code
+      vec_rank(NULL)
+    Condition
+      Error:
+      ! This type is not supported by `vec_order()`.
+
+---
+
+    Code
+      vec_rank(NULL, incomplete = "na")
+    Condition
+      Error:
+      ! This type is not supported by `vec_order()`.
```

Comment:
This used to throw the internal error seen in https://github.com/r-lib/vctrs/issues/1823

Reply:
Probably not worth it but would be nice if:

- Argument name was reported
- Argument value was revealed
- Error happened earlier and mentioned `vec_rank()`
- A `call` and `arg` argument would allow callers to take ownersip of the type checking